nservicebus/azure-service-bus/addressing-logic.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Azure Service Bus Transport Addressing Logic
+summary: Azure ServiceBus Transport Addressing Logic
+tags:
+- Azure
+- Cloud
+- Configuration
+---
+
+## Physical Addressing Logic
+
+One of the responsibilities of the transport is determining the names and physical location of entities in the underlying physical infrastructure. This is achieved by turning logical endpoint names into physical addresses of the Azure Service Bus entities, which is called *Physical Addressing Logic*.
+
+Prior to version 7 there was no explicit layer for handling this kind of logic, most assumptions such as assumptions on length limitations, legal characters, differences between paths and names etc... were sprinkled across the transport.
+
+Over time, subtle variations started to show in these assumptions as changes took place inside the implementation of certain namespace types. For example `Mixed` namespaces allowed paths and names up 290 characters, while `Messaging` namespaces are capped at 260. Next to that, customers also figured out ways to game the system (for valid reasons), like embedding `'/'` characters in endpoint names to create subfolders in the Azure Service Bus namespace registration system.
+
+### Version 6 and below
+
+In older versions some of these changes got exposed over time as lambda expressions and became collectively known as the [Naming Conventions](/nservicebus/azure-service-bus/naming-conventions.md#version-6-and-below-naming-conventions)
+
+### Version 7
+
+In version 7 the configuration API allows to modify all behaviours and assumptions related to the addressing logic. This article will mainly focus on describing the layers inside the addressing logic and will show how to replace parts if further changes should occur, for a full list of out of the box options refer to [Full Configuration API](/nservicebus/azure-service-bus/configuration/configuration.md).
+
+### Addressing layers
```

Comment:
Is the word "Layers" used in ASB in that context? It seems a bit surprising to me, because when I saw layers I thought about something like different levels of abstractions, e.g. physical vs logical address.